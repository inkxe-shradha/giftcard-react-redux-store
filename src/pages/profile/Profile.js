import {
  Card,
  Container,
  Grid,
  CardHeader,
  CardContent,
  Typography,
  Paper,
  Avatar,
  IconButton,
} from "@mui/material";
import { makeStyles } from "@mui/styles";
import PropTypes from "prop-types";
import React from "react";
import { connect } from "react-redux";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import { Link } from "react-router-dom";
const styles = makeStyles((theme) => ({
  userPaper: {
    padding: theme.spacing(2),
    border: "1px solid" + theme.palette.primary.main,
    wordBreak: "break-all",
  },
  avatar: {
    margin: "auto",
    backgroundColor: theme.palette.secondary.main,
  },
}));
export const Profile = ({ user }) => {
  const classes = styles();
  return (
    <Container maxWidth="lg" sx={{ marginTop: 10 }}>
      <Grid container spacing={2}>
        <Grid item xs={12} md={12}>
          <Paper
            elevation={12}
            sx={{
              padding: "2rem",
            }}
          >
            <Card>
              <CardHeader
                title="Profile Details"
                action={
                  <IconButton
                    aria-label="back to dashboard"
                    LinkComponent={Link}
                    color="inherit"
                    to="/"
                  >
                    <ArrowBackIcon />
                  </IconButton>
                }
              />
              <CardContent>
                <Grid container spacing={2}>
                  {/* Avatar Content */}
                  <Grid
                    item
                    xs={12}
                    margin={"auto"}
                    md={4}
                    alignContent="center"
                    justifyContent="center"
                  >
                    <Avatar
                      alt={user?.name}
                      src={user?.imageUrl}
                      sx={{ width: 200, height: 200 }}
                      className={classes.avatar}
                    />
                  </Grid>
                  {/* User Details */}
                  <Grid item xs={12} md={8}>
                    <div>
                      <Typography variant="h6">Name</Typography>
                      <Paper elevation={5} className={classes.userPaper}>
                        {user?.name}
                      </Paper>
                    </div>
                    <div>
                      <Typography variant="h6">Email</Typography>
                      <Paper elevation={5} className={classes.userPaper}>
                        {user?.email}
                      </Paper>
                    </div>
                    <div>
                      <Typography variant="h6">Balance</Typography>
                      <Paper elevation={5} className={classes.userPaper}>
                        {user?.balance_points || 0} <strong>$</strong>
                      </Paper>
                    </div>
                  </Grid>
                </Grid>
              </CardContent>
            </Card>
          </Paper>
        </Grid>
      </Grid>
    </Container>
  );
};

Profile.propTypes = {
  user: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  user: state.auth.user,
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
