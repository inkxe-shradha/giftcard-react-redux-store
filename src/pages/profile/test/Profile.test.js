import Profile from "../Profile";
import thunk from "redux-thunk";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { ThemeProvider } from "@mui/system";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { createTheme } from "@mui/material";
import { render, screen } from "@testing-library/react";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const theme = createTheme();
const state = {
  auth: {
    user: {},
    isAuthenticated: false,
  },
  loading: {
    loading: false,
  },
};
describe("Profile test suite", () => {
  const customRender = (location, initialState = state) => {
    const history = createMemoryHistory();
    history.push(location);
    const store = mockStore(initialState);
    return render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            <Profile />
          </Router>
        </ThemeProvider>
      </Provider>
    );
  };

  it("Should render the component snapshot", () => {
    const { asFragment } = customRender("/profile");
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should contain the passed user state object in the document", () => {
    customRender("/profile", {
      ...state,
      auth: {
        ...state.auth,
        user: {
          name: "John Doe",
          email: "jhonedoe@yahoo.com",
          imageUrl: "https://source.unsplash.com/random/900X700/?avatar",
          balance_points: 2000,
        },
      },
    });
    const nameElement = screen.getByText(/John Doe/i);
    const emailElement = screen.getByText(/jhonedoe@yahoo.com/i);
    const balanceElement = screen.getByText(/2000/i);
    expect(nameElement).toBeTruthy();
    expect(emailElement).toBeTruthy();
    expect(balanceElement).toBeTruthy();
  });
});
