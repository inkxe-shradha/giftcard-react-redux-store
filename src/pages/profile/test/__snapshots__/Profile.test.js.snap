// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile test suite Should render the component snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1dj8qml-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation12 css-nimraf-MuiPaper-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  Profile Details
                </span>
              </div>
              <div
                class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
              >
                <a
                  aria-label="back to dashboard"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  href="/"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowBackIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1dnd7uz-MuiGrid-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault makeStyles-avatar-2 css-1n3lyxl-MuiAvatar-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                      data-testid="PersonIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
                >
                  <div>
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      Name
                    </h6>
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 makeStyles-userPaper-1 css-qh7kl5-MuiPaper-root"
                    />
                  </div>
                  <div>
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      Email
                    </h6>
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 makeStyles-userPaper-1 css-qh7kl5-MuiPaper-root"
                    />
                  </div>
                  <div>
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      Balance
                    </h6>
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 makeStyles-userPaper-1 css-qh7kl5-MuiPaper-root"
                    >
                      0 
                      <strong>
                        $
                      </strong>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
